### ISSUE TITLE
Row hinting doesn't work

### ISSUE THREAD

**Issue by joshas** [MeanEYE/Sunflower#349]
In Preferences -> Item List -> Look & Feel section, option "Row hinting" doesn't work. set_rules_hint function is deprecated ( https://developer.gnome.org/gtk3/stable/GtkTreeView.html#gtk-tree-view-set-rules-hint ).
Also, set_rules_hint were used in Preferences TreeView controls, that doesn't work too.

**Comment by MeanEYE** [comment]
While it is deprecated I feel some people love and rely on this functionality. That said we should be able to replicate functionality through CSS. I'll test this in upcoming days. If CSS styling works, we will resort to setting class as opposed to setting this function.

**Comment by dajare** [comment]
> While it is deprecated I feel some people love and rely on this functionality. . . .

I've just "found" Sunflower FM and was fairly perplexed as to why my "preference" wasn't having any effect, being one of those people who "love" (if not exactly "rely") this functionality. 

Should this option at least be removed from the settings list if it's known not to work? Even better would be reinstating this feature somehow! Thanks. :)

(Using Sunflower FM 0.4 (62) on Ubuntu 18.04 LTS)

**Comment by MeanEYE** [comment]
It will be reinstated as we can define it with CSS. Functions are still there I think in the code, but for some reason list behaves the same way with and without. I personally don't use this feature so it took a while for someone to notice.

**Comment by dajare** [comment]
And thanks for mega-swift reply! Great to see this still under active development.

**Comment by MeanEYE** [comment]
You are welcome. Active is a bit of an overstatement. I try to find time here and there. It's usually grouped over the weekends when I have downtime from other obligations. However I am not going to abandon this project as it means too much to me.

**Comment by MeanEYE** [comment]
Unfortunately this is another feature GTK developers removed and made it impossible to bring back. You can see [more info here](https://gitlab.gnome.org/GNOME/gtk/-/issues/581). In short they think cost to benefit ratio is not good enough and no matter how much you like this feature they feel it's pointless.

Sadly there's very little I can do to solve this. They removed `row` and `column` elements from TreeView so I can't target styles properly. Every other widget does support this `:nth-child(odd)` pseudo-selector, except for TreeView. ListBox is more flexible and nicer than TreeView (also more modern) but issue with it is that it doesn't support tree expanders and it would use more memory I think.

I tried to come up with few different solutions but none worked to my satisfaction. They either make memory usage go up or are unreliable and make the whole program needlessly complex. If anyone has suggestions I am more than happy to listen to them.

**Comment by dajare** [comment]
Thanks for update and info. Perhaps this is a case of "close" == "can't fix"?

**Comment by MeanEYE** [comment]
To be honest I don't like the idea of doing that. GTK developers had caused
similar issues previously as well. For a while copy and paste didn't work
because they felt like not fixing a bug they caused.

I'll leave this open for a while at least in hopes we find a solution but
its unlikely.

**Comment by lee-b** [comment]
Sounds like the right thing is to mark this as blocked on a libgtk issue?

**Comment by MeanEYE** [comment]
I am set on resolving transition fast. But who knows. Right now it's like whack-a-mole.

**Comment by MeanEYE** [comment]
This feature was removed by GTK+ completely. At point I entertained a hope that I'll be able to re-implement it using `:odd` and `:even` CSS pseudo selectors.

**Comment by MeanEYE** [comment]
Yeah it seems this is not going to ever be possible again. Sorry folks.

[More info](https://gitlab.gnome.org/GNOME/gtk/-/issues/581#note_746153)

**Comment by lee-b** [comment]
> Yeah it seems this is not going to ever be possible again. Sorry folks.
> 
> [More info](https://gitlab.gnome.org/GNOME/gtk/-/issues/581#note_746153)

Fair enough, but the option should be disabled then, imho. It's confusing otherwise.
